components/Chips/src/MDCChipView.h

```diff
@@ -0,0 +1,169 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+@interface MDCChipView : UIControl
+
+/*
+ A UIImageView that leads the title label.
+
+ Only visible when the image is non-nil.
+ */
+@property(nonatomic, readonly, nonnull) UIImageView *imageView;
+
+/*
+ A UIImageView that leads the title label. Appears above the imageView.
+
+ Only visible when the chip is selected and the image is non-nil.
+ */
+@property(nonatomic, readonly, nonnull) UIImageView *selectedImageView;
+
+/*
+ A UIView that trails the title label.
+
+ It will be given a size based on the value returned from sizeThatFits:.
+ */
+@property(nonatomic, strong, nullable) UIView *accessoryView;
```

Comment:
Exposing a readwrite view property means that we are going to have to cleanup / rebuild the hierarchy in every dequeCell / prepareCellForReuse.  Is there any way we can do this without a writable view?

If we have selectedImageView, do we need selectedAccessoryView?
It would mimic UITableViewCell's editingAccessoryView.

Reply:
If you have any suggestions for other ways to handle this I'm all ears, but this appears to be the best compromise.

We don't need selectedAccessoryView. `selectedImageView` Serves a very specific purpose in reflecting what is described in the Chip designs.